### ISSUE TITLE
[a11y]: UI Shell [Side nav all variants] - current links not announced by SR

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15271]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.1

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
Current selection of the element is not announced by screen reader

Link is visually looking link selected but the screen reader is not announcing the same. 

### Recommended Fix
Screen reader should also announce about the selection of the elements.

People who uses screen reader will not be aware of the state of the element.

### WCAG 2.1 Violation
4.1.2 Name, Role, Value

**Comment by yujinpingping** [comment]
react version is: @carbon/react@1.42.1
